addons/hr_holidays/__init__.py

```diff
@@ -9,10 +9,16 @@
 
 from odoo import api, SUPERUSER_ID
 
+
+def _auto_install_l10n_hr_holidays(env):
+    country_codes = env['res.company'].search([]).mapped('country_id.code')
+    if not country_codes:
+        return
+    possible_module_names = [f'l10n_{country_code.lower()}_hr_holidays' for country_code in country_codes]
+    modules = env['ir.module.module'].search([('name', 'in', possible_module_names), ('state', '=', 'uninstalled')])
+    if modules:
+        modules.sudo().button_install()
+
+
 def _hr_holiday_post_init(env):
-    french_companies = env['res.company'].search_count([('partner_id.country_id.code', '=', 'FR')])
-    if french_companies:
-        env['ir.module.module'].search([
-            ('name', '=', 'l10n_fr_hr_work_entry_holidays'),
-            ('state', '=', 'uninstalled')
-        ]).sudo().button_install()
+    _auto_install_l10n_hr_holidays(env)
```

Comment:
It seems to me that this is all actually unnecessary. that just adding the countries in the manifest with a dependency on hr_holidays will also automatically install the l10n's on installing hr_holidays.